src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -388,54 +388,64 @@ def add_photon_series_to_nwbfile(
     """
 
     iterator_options = iterator_options or dict()
+    metadata = metadata or {}
 
-    metadata_copy = {} if metadata is None else deepcopy(metadata)
     assert photon_series_type in [
         "OnePhotonSeries",
         "TwoPhotonSeries",
     ], "'photon_series_type' must be either 'OnePhotonSeries' or 'TwoPhotonSeries'."
-    metadata_copy = dict_deep_update(
-        get_nwb_imaging_metadata(imaging, photon_series_type=photon_series_type), metadata_copy, append_list=False
-    )
-
-    if photon_series_type == "TwoPhotonSeries" and "OnePhotonSeries" in metadata_copy["Ophys"]:
```

Comment:
This should be valid. That is, we should be able to pass both types of metadata at the same time.

I think it was here to try to warn users that somehow they were passing their data wrong but then it has false positives. 

Once we switch to keys instead of indices to identify metadata entries we will be able to provide better errors to let user know they did not specify this well.